### PR TITLE
Changes to view all templates page and create new disruptions from templates

### PR DESCRIPTION
**Pull request by alvak1**
- Contains changes to view all templates page which has the common component to display data in sortable table
- Changes to create new disruption from template for staff and publisher type users

### CODE REVIEW

File: site/pages/api/duplicate-disruption.api.ts [hunk 1]
```diff
@@ -78,24 +92,24 @@ const duplicateDisruption = async (req: NextApiRequest, res: NextApiResponse): P
...
-            template === "true" ? ["template"] : [],
-            `${REVIEW_DISRUPTION_PAGE_PATH}/${newDisruptionId}`,
-            ["duplicate=true"],
+        const returnPath = encodeURIComponent(
+            `${DISRUPTION_DETAIL_PAGE_PATH}/${
+                templateId as string
+            }?template=true&return=${VIEW_ALL_TEMPLATES_PAGE_PATH}`,
         );
+        createDisruptionFromTemplate
+            ? redirectTo(res, `${CREATE_DISRUPTION_PAGE_PATH}/${newDisruptionId}?return=${returnPath}`)
```
Comment (alainajaffer) [inline site/pages/api/duplicate-disruption.api.ts:111]: you could use the redirect with query params function here

File: site/pages/review-disruption/review-disruption.test.tsx [hunk 2]
```diff
@@ -193,10 +203,127 @@ describe("pages", () => {
...
+            expect(header).toBeTruthy();
+            expect(cancelButton).toBeFalsy();
+
+            unmount();
+        });
+
+        it("should render correctly with appropriate buttons", () => {
+            // useRouter.mockImplementation(() => ({
+            //     query: {
+            //         return: `${DISRUPTION_DETAIL_PAGE_PATH}/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee?template=true&return=${VIEW_ALL_TEMPLATES_PAGE_PATH}`,
```
Comment (alainajaffer) [inline site/pages/review-disruption/review-disruption.test.tsx:246]: can this be removed?

File: site/pages/review-disruption/review-disruption.test.tsx [hunk 3]
```diff
@@ -193,10 +203,127 @@ describe("pages", () => {
...
+            expect(deleteTemplateButton).toBeFalsy();
+            expect(cancelButton).toBeTruthy();
+
+            unmount();
+        });
+
+        it("should render correctly with appropriate buttons for staff user", () => {
+            // useRouter.mockImplementation(() => ({
+            //     query: {
+            //         return: `${DISRUPTION_DETAIL_PAGE_PATH}/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee?template=true&return=${VIEW_ALL_TEMPLATES_PAGE_PATH}`,
```
Comment (alainajaffer) [inline site/pages/review-disruption/review-disruption.test.tsx:289]: can we remove this?

File: site/pages/view-all-templates.page.tsx [hunk 4]
```diff
@@ -1,53 +1,39 @@
...
-const ViewAllTemplates = ({ newDisruptionId }: ViewAllTemplatesProps): ReactElement => {
+const ViewAllTemplates = ({
+    newContentId,
+    adminAreaCodes,
+    filterStatus,
+    enableLoadingSpinnerOnPageLoad = true,
+}: ViewAllContentProps): ReactElement => {
     return (
         <BaseLayout title={title} description={description}>
-            <h1 className="govuk-heading-xl">View all templates</h1>
```
Comment (alainajaffer) [inline site/pages/view-all-templates.page.tsx:19]: the header says view all disruptions i think it should be view all templates

File: site/pages/api/duplicate-disruption.api.ts (hunk 1 above)
Comment (alvak1) [reply site/pages/api/duplicate-disruption.api.ts]: done

File: site/pages/review-disruption/review-disruption.test.tsx (hunk 2 above)
Comment (alvak1) [reply site/pages/review-disruption/review-disruption.test.tsx]: done

File: site/pages/review-disruption/review-disruption.test.tsx (hunk 3 above)
Comment (alvak1) [reply site/pages/review-disruption/review-disruption.test.tsx]: done

File: site/pages/view-all-templates.page.tsx (hunk 4 above)
Comment (alvak1) [reply site/pages/view-all-templates.page.tsx:19]: done

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**